drivers/thermal/gov_step_wise.c

```diff
@@ -99,12 +99,15 @@ static void thermal_zone_trip_update(struct thermal_zone_device *tz, int trip_id
 	}
 
 	hyst_temp = trip_temp = trip.temperature;
-	if (tz->ops->get_trip_hyst) {
-		tz->ops->get_trip_hyst(tz, trip_id, &hyst_temp);
-		hyst_temp = trip_temp - hyst_temp;
-	}
 	trip_type = trip.type;
 
+	if (trip.hysteresis)
```

Comment:
I don't see why you've put this code here - the location of the old version is more logical.